docs/templates/feature.md

```diff
@@ -51,27 +50,77 @@ Potential *offline* types are:
+If users query directly the online feature store, for instance using the `fs.sql("SELECT ...", online=True)` statement, they will receive back a binary blob.
+
+On the feature store UI, the online feature type for complex features will be reported as *VARBINARY*.
+
+#### Online restrictions for primary key types:
+
+When a feature is being used as a primary key, certain types are not allowed. Examples of such types are *Float*, *Double*, *Date*, *Text*, *Blob* and *Complex Types*  (e.g. Array<>). Additionally the size of the sum of the primary key online types storage requirements should not exceed 3KB.
+
+### Type Inference
+
+The offline and online types for each feature are inferred automatically when calling the [save()](../api/feature_group_api/#save) method to create a feature group. The types will be inferred based on the types of the Spark or Pandas DataFrame.
+
+In the case of Spark DataFrame, the [Spark types](https://spark.apache.org/docs/latest/sql-ref-datatypes.html) will be converted in the equivalent Hive Metastore type and used as offline feature store type. If the feature group is online enabled, Hopsworks will then match the offline type to the corresponding online type. The matching is performed as following:
+
+* If the offline type is supported also on the online feature store (e.g. INT, FLOAT, DATE, TIMESTAMP), the online type will be equivalent to the offline type
+* If the offline type is *boolean*, the online type is going to be set as *tinyint*
+* If the offline type is *string*, the online type is going to be set as *varchar(100)*
+* If the offline type is not supported by the online feature store and it is not one of the above exception, the online type will be set as *varbinary(100)* to handle complex types.
+
+#### Pandas Conversion
+
+When registering a [Pandas](https://pandas.pydata.org/) DataFrame as feature group, the following conversion is applied:
+
+| Pandas Type        | Offline Feature Type|
+| ------------------ | ------------------- |
+| int32              | INT                 |
+| int64              | BIGINT              |
+| float32            | FLOAT               |
+| float64            | DOUBLE              |
+| datetime64[ns]     | TIMESTAMP           |
+| object             | STRING              |
+
+
+### Type Overwrite
+
+When creating a feature group it is possible for the user to control both the offline and online type of each feature. If users manually specify a schema for the feature group, Hopsworks is going to use it to create the feature group, without performing any type inference.
```

Comment:
feature -> column
manually -> explicitly define the schema
use it -> use that schema

type inference is ok here, i think. You could also say - without automatically mapping the user defined schema to the offline and online data types